auth.go

```diff
@@ -385,7 +380,24 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 	if eGetIdent != nil {
 		return nil, ErrIdentityAuthNotFound
 	}
-	if eAuth := x.authenticator.Authenticate(identity, password); eAuth == nil {
+	user, eGetUser := x.userStore.GetUser(userId)
+	if eGetUser != nil {
+		return nil, ErrIdentityAuthNotFound
+	}
+	var eAuth error
+	if user.Type == LDAPUserType {
```

Comment:
Why are we calling Authenticate(identity, password), when we always go to the trouble of first fetching the UserID. Surely we should make a function called Authenticate(userID, password). That would remove the need for all callers of Authenticate to code this special logic about turning ErrInvalidCredentials into ErrInvalidPassword.